### PR TITLE
Fix issue#1

### PR DESCRIPTION
**Pull request by Abhishek-Punhani**
backend/deck.js
Fixed the function to initialize the deck according to the requirements and added valid checks to check the composition of deck

Fixes: #1

### CODE REVIEW

File: backend/uno-game-engine/deck.ts [hunk 1]
```diff
@@ -74,10 +92,7 @@ export function makeCard(
  * This function shuffles the elements of the given array *in place* . The function behaves in a type-agnostic way.
  * Time complexity: O(n)
  */
-export function shuffle(deck: Array<UNOCard>) {
-    // Fisher-Yates shuffle algorithm to shuffle card deck
-    for (let i = deck.length - 1; i > 0; i--) {
-        const j = Math.floor(Math.random() * i);
-        [deck[i], deck[j]] = [deck[j], deck[i]];
-    }
+export function shuffle(deck: Array<any>) {
```
Comment (kuv2707) [inline backend/uno-game-engine/deck.ts:95]: You seem to accidentally have removed a changed added through a previous PR.

Comment (Abhishek-Punhani) [reply backend/uno-game-engine/deck.ts]: Bhaiya I have fixed this issue and included the shuffle function as prescribed 
Kindly review now

File: backend/uno-game-engine/deck.ts [hunk 2]
```diff
@@ -14,40 +15,57 @@ const values = [
...
+            }
+        }
+    }
+
+    // generating special cards
+    for (let sc of specialCards) {
+        for (let i = 0; i < 4; i++) {
+            deck.push(makeCard('wild', 'wild', sc));
+        }
+    }
```
Comment (AadityanshuSingh) [inline backend/uno-game-engine/deck.ts:68]: The logic is good to go.

**Comment by kuv2707** [comment]
@Abhishek-Punhani Thanks for working on this, but another PR has been merged, closing the issue.

**Pull request closed without merging**